File: eBank.DataAccess/Models/Account/AccountModel.cs
using eBank.DataAccess.Models.Base;
using eBank.DataAccess.Models.User;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eBank.DataAccess.Models.Account
{
    public class AccountModel : BaseModel
    {
        [Key]
        public int AccountId { get; set; }

        public long AccountNumber { get; set; }

        public int AccountType { get; set; }

        public double Balance { get; set; }

        public bool AccountStatus { get; set; }

        public string Id { get; set; }

        [ForeignKey("Id")]
        public ApplicationUser User { get; set; }

        [NotMapped]
        public string MaskedAccountNumber { get; set; }
    }
}

File: eBank.Web/ClientApp/src/app/account/models/account.model.ts
import { IAccount } from '../interfaces/account.interface';

export class Account implements IAccount {
    accountId?: number;
    accountNumber: number;
    accountType: number;
    balance: number;
    accountStatus: boolean;
    id: string;
    parentId?: number;
    maskedAccountNumber: string;
}
